src/API/OnboardingTasks.php

```diff
@@ -648,237 +671,68 @@ public function get_status() {
 	 * @return WP_REST_Response|WP_Error
 	 */
 	public function get_tasks() {
-		$profiler_data         = get_option( Onboarding::PROFILE_DATA_OPTION, array() );
-		$installed_plugins     = PluginsHelper::get_installed_plugin_slugs();
-		$product_types         = isset( $profiler_data['product_types'] ) ? $profiler_data['product_types'] : array();
-		$allowed_product_types = Onboarding::get_allowed_product_types();
-		$purchaseable_products = array();
-		$remaining_products    = array();
-		foreach ( $product_types as $product_type ) {
-			if ( ! isset( $allowed_product_types[ $product_type ]['slug'] ) ) {
-				continue;
-			}
+		$task_lists = OnboardingTasksFeature::get_task_lists();
+		return rest_ensure_response( $task_lists );
+	}
+
+	/**
+	 * Dismiss a single task.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Request|WP_Error
+	 */
+	public function dismiss_task( $request ) {
+		$id = $request->get_param( 'id' );
 
-			$purchaseable_products[] = $allowed_product_types[ $product_type ];
+		$is_dismissable = false;
 
-			if ( ! in_array( $allowed_product_types[ $product_type ]['slug'], $installed_plugins, true ) ) {
-				$remaining_products[] = $allowed_product_types[ $product_type ]['label'];
+		foreach ( OnboardingTasksFeature::get_task_lists() as $task_list ) {
+			foreach ( $task_list['tasks'] as $task ) {
+				if ( $id === $task['id'] && isset( $task['isDismissable'] ) && $task['isDismissable'] ) {
```

Comment:
I think it would be worth creating a `get_task` method on the `OnboardingTasksFeature`, and move this logic to there (aside from the `isDismissable` stuff).
As retrieving an individual task might be common, and having to write this double loop every time is not great.

Reply:
@louwie17 Definitely agree.  @joelclimbsthings has some good ideas in https://github.com/woocommerce/woocommerce-admin/pull/7539 so I'll add a todo note to use his implementation once merged.